models/type/src/field-type/transformers.ts

```diff
@@ -0,0 +1,90 @@
+import { Reference } from '@commercetools-test-data/commons';
+import {
+  Transformer,
+  buildField,
+  buildFields,
+} from '@commercetools-test-data/core';
+import { faker } from '@faker-js/faker';
+import * as CustomFieldEnumValue from '../custom-field-enum-value';
+import { TCustomFieldEnumValue } from '../custom-field-enum-value';
+import * as CustomFieldLocalizedEnumValue from '../custom-field-localized-enum-value';
+import { TCustomFieldLocalizedEnumValue } from '../custom-field-localized-enum-value';
+import FieldType from './builder';
+import { typeNames, graphqlTypenameByFieldTypeName } from './constants';
+import { TFieldName, TFieldType, TFieldTypeGraphql } from './types';
+
+const list = <T>(fn: (index?: number) => T) =>
+  Array.from({ length: faker.number.int({ min: 1, max: 7 }) }, (_, index) =>
+    fn(index)
+  );
+
+const resolveCommonTransformations = <T extends TFieldType | TFieldTypeGraphql>(
+  fields: TFieldType
+): T => {
+  const commonFields = {
+    name: fields.name as TFieldName,
+    __typename: graphqlTypenameByFieldTypeName[fields.name],
```

Comment:
This property should not be common to all transformers, right? I think it's supposed to only exists in the GraphQL representation.